Net12/WebMaze/Controllers/MazeController.cs

```diff
@@ -1,33 +1,67 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
 using Net12.Maze;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using WebMaze.EfStuff;
+using WebMaze.EfStuff.DbModel;
+using WebMaze.EfStuff.Repositories;
 using WebMaze.Models;
 
 namespace WebMaze.Controllers
 {
     public class MazeController : Controller
     {
-      
+        private MazeDifficultRepository _mazeDifficultRepository;
+        private IMapper _mapper;
+
+        public MazeController(MazeDifficultRepository mazzeDifficultRepository, IMapper mapper)
+        {
+            _mazeDifficultRepository = mazzeDifficultRepository;
+            _mapper = mapper;
+        }
+
         public IActionResult Index(int width, int height)
         {
             var mazeBuilder = new MazeBuilder();
             var maze = mazeBuilder.Build(width, height, 50, 100, true);
             return View(maze);
         }
 
-        public IActionResult Trap()
+        [HttpGet]
+        public IActionResult AddMazeDifficult()
         {
-            var model = new CellInfoViewModel();
+            return View();
+        }
+
+        [HttpPost]
+        public IActionResult AddMazeDifficult(MazeDifficultProfileViewModel mazeDifficultProfileViewModel)
+        {
+            var dbMazeDifficult = _mapper.Map<MazeDifficultProfile>(mazeDifficultProfileViewModel);
+            dbMazeDifficult.IsActive = true;
+            dbMazeDifficult.Creater = _mazeDifficultRepository.GetRandomUser();
```

Comment:
You can use the UserRepository method "GetRandomUser" here. Add the parametr of this type and write it in the constructor.